src/freenet/clients/fcp/FCPPluginServerMessage.java

```diff
@@ -0,0 +1,183 @@
+ * Public License, version 2 (or at your option any later version). See
+ * http://www.gnu.org/ for further details of the GPL. */
+package freenet.clients.fcp;
+
+import freenet.node.Node;
+import freenet.pluginmanager.FredPluginFCPMessageHandler.FCPPluginMessage;
+import freenet.pluginmanager.PluginManager;
+import freenet.support.SimpleFieldSet;
+import freenet.support.api.Bucket;
+
+/**
+ * This class produces the network format for a FCP message which is send from a FCP server
+ * plugin to a FCP client.<br>
+ * It is the inverse of {@link FCPPluginClientMessage} which parses the on-network format of client
+ * to server messages.<br><br>
+ * 
+ * There is a similar class {@link FCPPluginMessage} which serves as a container of FCP plugin
+ * messages which are produced and consumed by the actual server and client plugin implementations.
+ * Consider this class here as an internal representation of FCP plugin messages used solely
+ * for encoding server-to-client messages, while the other one is the external representation used
+ * for both server and client messages.
+ * 
+ * ATTENTION: The on-network name of this message is different: It is {@value #NAME}. The class
+ * previously had the same name but it was decided to rename it: Previously, it was only allowed
+ * for the server to send messages to the client as a direct <i>reply</i> to a message. Nowadays,
+ * the server can send messages to the client any time he wants, even if there hasn't been a client
+ * message for a long time. So the class was renamed to FCPPluginServerMessage to prevent the
+ * misconception that it would be reply only, and also to make the name symmetrical to class
+ * {@link FCPPluginClientMessage}. To stay backward compatible, it was decided to keep the raw
+ * network message name as is. <br>
+ * TODO: Would it technically be possible to add a second name to the on-network data so we
+ * can get rid of the old name after a transition period?<br><br>
+ * 
+ * @link FCPPluginClient FCPPluginClient gives an overview of the code paths which messages take.
+ * @author saces
+ * @author xor (xor@freenetproject.org)
+ */
+public class FCPPluginServerMessage extends DataCarryingMessage {
+	
+    /**
+     * On-network format name of the message.
+     * 
+     * ATTENTION: This one is different to the class name. For an explanation, see the class-level
+     * JavaDoc {@link FCPPluginServerMessage}.
+     */
+	private static final String NAME = "FCPPluginReply";
+	
+    public static final String PARAM_PREFIX = "Replies";
+
+    /** @see FCPPluginMessage#data */
+	private final long dataLength;
+
+    /** @see FCPPluginClient#getServerPluginName() */
+	private final String plugname;
+
+    /** @see FCPPluginMessage#identifier */
+	private final String identifier;
+
+    /** @see FCPPluginMessage#params */
+	private final SimpleFieldSet plugparams;
+
+    /** @see FCPPluginMessage#success */
+    private final Boolean success;
```

Comment:
Look more closely: It is explained at FCPPluginMessage.success, which is also linked by the JavaDoc.